### ISSUE TITLE
Force authentication on swagger API?

### ISSUE THREAD

**Issue by djMax** [swagger-api/swagger-js#1266]
The 2.x client must've just blindly sent authorizations somehow, because now upgrading to 3, a spec that used to work doesn't anymore because it won't send basic auth headers. The spec has "security" at the root specifying that it wants basic, but I don't think swagger-client cares about that, it only cares about the operation spec. How can I force auth without just doing it myself in requestInterceptor?

**Comment by shockey** [comment]
@djMax can you share an example API definition? Kinda hard to follow what you're describing in the abstract.

**Comment by djMax** [comment]
I can't follow it in the code, but turns out I'm wrong, it does take the security key of the root of the swagger spec and apply it to all operations. I just had the format wrong originally. So if you put:

`"security":[{"basic":[]}]`

At root, it will apply basic to all calls.
